bundler/lib/bundler/cli.rb

```diff
@@ -815,7 +815,10 @@ def remembered_flag_deprecation(name)
       option = current_command.options[name]
       flag_name = option.switch_name
 
-      name_index = ARGV.find {|arg| flag_name == arg }
+      name_index = ARGV.find do |arg|
+        flag_name == arg || arg.split("=").find {|str| flag_name == str }
+      end
+
```

Comment:
This implementation is subject to pathological false positives like

```
$ mkdir ./--path
$ bundle install --binstubs=--path
[DEPRECATED] The `--path` flag is deprecated because it relies on being remembered across bundler invocations, which bundler will no longer do in future versions. Instead please use `bundle config set path 'path'`, and stop using this flag
[DEPRECATED] The --binstubs option will be removed in favor of `bundle binstubs`
Could not locate Gemfile
```

I believe you can go with the simpler

```ruby
      name_index = ARGV.find {|arg| flag_name == arg.split("=")[0] }
```